runtime/sema/authaccount_contracts.go

```diff
@@ -0,0 +1,253 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2020 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sema
+
+import (
+	"github.com/onflow/cadence/runtime/ast"
+	"github.com/onflow/cadence/runtime/common"
+)
+
+// AuthAccountContractsType represents the type `AuthAccount.Contracts`
+
+type AuthAccountContractsType struct{}
+
+func (*AuthAccountContractsType) IsType() {}
+
+func (*AuthAccountContractsType) String() string {
+	return "Contracts"
+}
+
+func (*AuthAccountContractsType) QualifiedString() string {
+	return "AuthAccount.Contracts"
+}
+
+func (*AuthAccountContractsType) ID() TypeID {
+	return "AuthAccount.Contracts"
+}
+
+func (*AuthAccountContractsType) Equal(other Type) bool {
+	_, ok := other.(*AuthAccountContractsType)
+	return ok
+}
+
+func (*AuthAccountContractsType) IsResourceType() bool {
+	return false
+}
+
+func (*AuthAccountContractsType) IsInvalidType() bool {
+	return false
+}
+
+func (*AuthAccountContractsType) IsStorable(_ map[*Member]bool) bool {
+	return false
+}
+
+func (*AuthAccountContractsType) IsEquatable() bool {
+	// TODO:
```

Comment:
Noting in case you wanted to address in this PR

Reply:
Thanks for flagging, it's just a future potential TODO
```suggestion
	// TODO: maybe implement equality
```